From 3e4847da90c8ba9561b7988a865b92752bda4cbc Mon Sep 17 00:00:00 2001
From: Yinbo Zhu <yinbo.zhu@nxp.com>
Date: Wed, 5 Dec 2018 14:15:11 +0800
Subject: [PATCH 384/706] arm64: dts: ls1028a: Add property
 "snps,incr-burst-type-adjustment"

Property "snps,incr-burst-type-adjustment = <x>, <y>..." for USB3.0 DWC3.
When only one value means INCRx mode with fix burst type.
When more than one value, means undefined length burst mode, USB controller
can use the length less than or equal to the largest enabled burst length.

While enabling undefined length INCR burst type and INCR16 burst type,
get better write performance on NXP Layerscape platforms:
around 3% improvement (from 364MB/s to 375MB/s).

Signed-off-by: Yinbo Zhu <yinbo.zhu@nxp.com>
(cherry picked from commit 1fad3a7f9ca3bc94eddac53eb69509905e94bfc6)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 1773594b4793..8fd15bfab528 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -367,6 +367,7 @@
 			interrupts= <0 80 0x4>;
 			dr_mode= "host";
 			snps,dis_rxdet_inp3_quirk;
+			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
 		};
 
 		usb1:usb3@3110000 {
@@ -375,6 +376,7 @@
 			interrupts= <0 81 0x4>;
 			dr_mode= "host";
 			snps,dis_rxdet_inp3_quirk;
+			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
 		};
 
 		i2c0: i2c@2000000 {
-- 
2.17.1

